controller/server/main.go

```diff
@@ -138,6 +137,59 @@ func validatePath(path string) (string, error) {
 	return path, nil
 }
 
+func (s *server) CreateTopology(ctx context.Context, req *cpb.CreateTopologyRequest) (*cpb.CreateTopologyResponse, error) {
+	log.Infof("Received CreateTopology request: %v", req)
+	bp, err := validatePath(req.Topology.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid topology path: %v", err)
+	}
+	p := topo.TopologyParams{
+		TopoName:       req.Topology.Name,
+		TopoNewOptions: []topo.Option{topo.WithBasePath(filepath.Dir(bp))},
+		Kubecfg:        req.Kubecfg,
+	}
+	err = topo.CreateTopology(ctx, p)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to create topology: %v", err)
+	}
+
+	return &cpb.CreateTopologyResponse{
+		TopologyName: req.Topology.GetName(),
+		State:        cpb.TopologyState_TOPOLOGY_STATE_RUNNING,
+	}, nil
+
+}
+
+func (s *server) DeleteTopology(ctx context.Context, req *cpb.DeleteTopologyRequest) (*cpb.DeleteTopologyResponse, error) {
+	log.Infof("Received DeleteTopology request: %v", req)
+	p := topo.TopologyParams{
+		TopoName: req.TopologyName,
+	}
+	err := topo.DeleteTopology(ctx, p)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to delete topology: %v", err)
+	}
+
+	return &cpb.DeleteTopologyResponse{}, nil
+
+}
+
+func (s *server) ShowTopology(ctx context.Context, req *cpb.ShowTopologyRequest) (*cpb.ShowTopologyResponse, error) {
+	log.Infof("Received ShowTopology request: %v", req)
+	p := topo.TopologyParams{
+		TopoName: req.TopologyName,
+	}
+	tpb, err := topo.GetTopologyServices(ctx, p)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to show topology: %v", err)
+	}
+
+	return &cpb.ShowTopologyResponse{
+		State:    cpb.TopologyState_TOPOLOGY_STATE_RUNNING,
```

Comment:
so this feels like we are giving a state which we didn't actually check we just assume?

Reply:
agreed, there needs to be some code here that can check the status of a topology and classify which of the enum states the topo is in

Reply:
Added a check to derive the topology state with the node's pod state.